src/test/resources/local-pipeline/run_local_pipeline

```diff
@@ -0,0 +1,167 @@
+    docker run -it -d --name $CONTAINER_NAME_INPUTDOWNLOADER -v saps-vol:$INPUTDOWNLOADER_SHARED_FOLDER $IMAGE_NAME_INPUTDOWNLOADER
+    exit_handler
+  fi
+}
+
+function build_container_preprocessor {
+  if [  $IMAGE_NAME_PREPROCESSOR ]
+  then
+    #Code of the build container preprocessor
+    exit_handler
+  fi
+}
+
+function build_container_worker {
+  if [ $IMAGE_NAME_WORKER ]
+  then
+    #Check if worker container already exists
+    docker ps -a | grep "$CONTAINER_NAME_WORKER"
+    if [ $? -eq 0 ]
+    then
+      #Stop and remove existing worker container
+      docker stop $CONTAINER_NAME_WORKER
+      docker rm $CONTAINER_NAME_WORKER
+    fi
+    #Build a worker container from your image using volume for shared folder between host (saps-vol) and container (WORKER_SHARED_FOLDER)
+    docker run -it -d --name $CONTAINER_NAME_WORKER -v saps-vol:$WORKER_SHARED_FOLDER $IMAGE_NAME_WORKER
+    exit_handler
+  fi
+}
+
+function run_input_download {
+  if [ $IMAGE_NAME_INPUTDOWNLOADER ]
+  then
+    #Create directory of input, results and metadata before execute inputdownloader container
+    docker exec $CONTAINER_NAME_INPUTDOWNLOADER mkdir -p $INPUTDOWNLOADER_SHARED_FOLDER/$IMAGE_DATA_FOLDER /home/ubuntu/results /home/ubuntu/metadata
+    #Execute run script in inputdownloader container for the download and preprocess of the landsat image
+    EXIT_CODE_INPUTDOWNLOADER=`docker exec -e USGS_USERNAME=$USGS_USERNAME -e USGS_PASSWORD=$USGS_PASSWORD $CONTAINER_NAME_INPUTDOWNLOADER sh /home/ubuntu/run.sh $IMAGE_DATASET $IMAGE_REGION $IMAGE_DATE $INPUTDOWNLOADER_SHARED_FOLDER/$IMAGE_DATA_FOLDER $INPUTDOWNLOADER_SHARED_FOLDER/$IMAGE_DATA_FOLDER`
+  fi
+}
+
+function run_preprocessor {
+  if [ $IMAGE_NAME_PREPROCESSOR ]
+  then
+    #Code of the run preprocessor
+    EXIT_CODE_PREPROCESSOR=0
+  fi
+}
+
+function run_worker {
+  if [ $IMAGE_NAME_WORKER ]
+  then
+    #Move directory results and metadata for tmp (share directory)
+    docker exec $CONTAINER_NAME_INPUTDOWNLOADER mv /home/ubuntu/results /home/ubuntu/metadata $WORKER_SHARED_FOLDER
+    #Create process directory
+    docker exec $CONTAINER_NAME_WORKER mkdir -p $WORKER_SHARED_FOLDER/$PREPROCESS_FOLDER
+    #Execute run script in the worker container to generate results of the landsat image
+    EXIT_CODE_WORKER=`docker exec $CONTAINER_NAME_WORKER bash /home/ubuntu/bin/run.sh $WORKER_SHARED_FOLDER/$IMAGE_DATA_FOLDER $WORKER_SHARED_FOLDER/results $WORKER_SHARED_FOLDER/$PREPROCESS_FOLDER $WORKER_SHARED_FOLDER/metadata`
+  fi
+}
+
+function exit_handler {
+  if [ $? -ne 0 ]
+  then
+    exit 1
+  fi
+}
+
+check_docker
+
+pull_image_inputdownloader
+build_container_inputdownloader
+
+pull_image_preprocessor
+build_container_preprocessor
+
+pull_image_worker
+build_container_worker
+
+run_input_download
+run_preprocessor
+run_worker
+
+echo $EXIT_CODE_INPUTDOWNLOADER $EXIT_CODE_PREPROCESSOR $EXIT_CODE_WORKER
```

Comment:
almost done, @ThiagoWhispher 
i would favor returning the code in the run_* methods; so, instead of using a global $EXIT_CODE_* use a local variable in this way: `exit_code_inputdownloader=$(run_input_download)

Also, now, it seems better to me to have three code blocks: pull, build and run (inputdownload), then pre-proc, then worker ...

Reply:
@thiagomanel , then:
- pull_image_x, build_container_x and run_x, would be moved to a function called x (e.g. inputdownloader) and the same thing to others?
- the other modification would be something like: `local exit_code_inputdownloader=$ (inputdownloader), where inputdownloader is the new function mentioned above?